ambry-clustermap/src/main/java/com/github/ambry/clustermap/AmbryPartition.java

```diff
@@ -193,18 +193,32 @@ void resolvePartitionState() {
     // if the lock could not be taken, that means the state is being updated in another thread. Avoid updating the
     // state in that case.
     if (currentCounterValue > lastUpdatedSealedStateChangeCounter && stateChangeLock.tryLock()) {
+      ReplicaSealStatus mergedReplicaSealStatus = ReplicaSealStatus.NOT_SEALED;
       try {
         lastUpdatedSealedStateChangeCounter = currentCounterValue;
-        boolean isSealed = false;
+        AmbryReplica partiallySealedReplica = null;
+        AmbryReplica firstSealedReplica = null;
         for (AmbryReplica replica : clusterManagerQueryHelper.getReplicaIdsForPartition(this)) {
-          if (replica.isSealed()) {
-            LOGGER.trace("Partition {} is sealed because of replica {} is sealed at sealedStateChangeCounter {}", id,
-                replica.getDataNodeId().toString(), lastUpdatedSealedStateChangeCounter);
-            isSealed = true;
+          mergedReplicaSealStatus =
+              ReplicaSealStatus.mergeReplicaSealStatus(mergedReplicaSealStatus, replica.getSealedStatus());
+          if (mergedReplicaSealStatus == ReplicaSealStatus.SEALED) {
+            firstSealedReplica = replica;
             break;
           }
+          if (mergedReplicaSealStatus == ReplicaSealStatus.PARTIALLY_SEALED) {
+            partiallySealedReplica = replica;
+          }
+        }
+        if (mergedReplicaSealStatus == ReplicaSealStatus.SEALED) {
```

Comment:
I think we don't need firstSealedReplica variable and this if condition. 
This if condition is used to print the trace only.
We can just do it in the 

```
          if (mergedReplicaSealStatus == ReplicaSealStatus.SEALED) {
            firstSealedReplica = replica; <-- remove
            <-- add print here.
            break;
          }
```